src/core/public/locale_helper.ts

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * Extracts the locale value and parameter from a given URL string.
+ *
+ * @param url - The full URL string to parse
+ * @returns An object with localeValue, localeParam or null if not found, and updatedUrl or url if no updates.
+ */
+export function extractLocaleInfo(
+  url: string
+): { localeValue: string | null; localeParam: string | null; updatedUrl: string } {
+  const patterns = [
+    /[#&?](i18n-locale)=([^&/]+)/i, // Standard query parameter
+    /#\/&(i18n-locale)=([^&/]+)/i, // After hash and slash
+    /\/&(i18n-locale)=([^&/]+)/i, // After path and slash
+  ];
+
+  for (const pattern of patterns) {
+    const match = url.match(pattern);
```

Comment:
can this do something like https://github.com/opensearch-project/OpenSearch-Dashboards/blob/75d67c06e72cf157368e5175cb44bc54c16607af/src/core/public/chrome/chrome_service.tsx#L141
instead of regex?

Reply:
will try this one.

Reply:
Oh I remember why I switch to regex. The searchParams property of the URL object is designed to work with standard query parameters, which are typically preceded by a question mark (?) in the URL.
For the URL like `http://localhost:5601/app/home/&i18n-locale-zh-CN`, it doesn't follow the standard format for query parameters.The & is being used as if it's starting a query string, but without a preceding ?.The searchParams approach would not work directly and searchParams.get('i18n-locale') will return null. I think this is the same issue as our later discussion. 

If we always have standard query parameters then we are good to switch to searchParams. But if in the future we want to translate urls like our current home page `http://localhost:5603/xqy/app/home#/` or management `http://localhost:5603/xqy/app/management` then we can't use this. I am trying to make this feature applicable for future extension.

Reply:
I think it's answered in https://github.com/opensearch-project/OpenSearch-Dashboards/pull/7686#discussion_r1715682670
> When this feature exposes to users we should advertise it as "add a URL parameter of i18n-locale to change locale", not "append `&i18n-locale` to change locale", so i don't think we need to handle invalid URL parameters